Invalid upload request for random NPM packages

Hi, I was trying to use this plugin and it seemed to work at first.

But after some time it starts to fail during `npm install` commands in different unrelated projects.

Basically it responds with error from NPM:
```
npm ERR! code E500
npm ERR! 500 Internal Server Error: ms@0.7.2
```

And on server verdaccio prints:
```
verdaccio-gc |  http --> 200, req: 'GET https://registry.npmjs.org/ms' (streaming)
verdaccio-gc |  http --> 200, req: 'GET https://registry.npmjs.org/ms', bytes: 0/26050
verdaccio-gc |  error-=- gcloud: save package ms has failed err: Invalid Upload Request
verdaccio-gc |  http <-- 500, user: null(192.168.176.1), req: 'GET /ms', error: Invalid Upload Request
```

I'm using verdaccio as a docker container with GCloud.

At first I thought it was networking issue in our cluster but then I tested it on local machine with different google account and same errors were occurring.

The error happens for random NPM packages and once it hits this error it always responds with this error for that package. And if I'm installing other packages that do not have this error then everything works fine.

Also if I'm just navigating to my registry website and checking package with error (ex `http://localhost:4873/ms`) then I see response:
```
{
  "error": "Invalid Upload Request"
}
```

For other working packages it responds with proper JSON with versions...

I would like to help debug this. It would be great to know how can I enable more logging so we can have better idea.

--- a/README.md
+++ b/README.md
@@ -53,7 +53,14 @@ store:
    ## default validation is, it can be overrided by 
    ## https://cloud.google.com/nodejs/docs/reference/storage/1.6.x/File.html#createWriteStream
    # validation: crc32c
+
+   ## Enable/disable resumable uploads to GC Storage
+   ## By default it's enabled in `@google-cloud/storage`
+   ## May cause failures for small package uploads so it is recommended to set it to `false`
+   ## @see https://stackoverflow.com/questions/53172050/google-cloud-storage-invalid-upload-request-error-bad-request
+   resumable: true
 ```
+
 Define `env` whether you want load the value from environment variables.
 
 > If you are willing to use some of `env` just **do not define** properties on

--- a/src/storage.ts
+++ b/src/storage.ts
@@ -158,6 +158,11 @@ class GoogleCloudStorageHandler implements IPackageStorageManager {
       try {
         await file.save(this._convertToString(metadata), {
           validation: this.config.validation || defaultValidation,
+          /**
+           * When resumable is `undefined` - it will default to `true`as per GC Storage documentation:
+           * `Resumable uploads are automatically enabled and must be shut off explicitly by setting options.resumable to false`
+           * @see https://cloud.google.com/nodejs/docs/reference/storage/2.5.x/File#createWriteStream
+           */
           resumable: this.config.resumable
         });
         resolve(null);

--- a/src/storage.ts
+++ b/src/storage.ts
@@ -157,7 +157,10 @@ class GoogleCloudStorageHandler implements IPackageStorageManager {
       const file = this._buildFilePath(name, pkgFileName);
       try {
         await file.save(this._convertToString(metadata), {
-          validation: this.config.validation || defaultValidation
+          validation: this.config.validation || defaultValidation,
+          // Disable resumable uploads to avoid random? failures from google cloud
+          // @see https://stackoverflow.com/questions/53172050/google-cloud-storage-invalid-upload-request-error-bad-request
+          resumable: false
         });
         resolve(null);
       } catch (err) {

--- a/src/storage.ts
+++ b/src/storage.ts
@@ -158,9 +158,7 @@ class GoogleCloudStorageHandler implements IPackageStorageManager {
       try {
         await file.save(this._convertToString(metadata), {
           validation: this.config.validation || defaultValidation,
-          // Disable resumable uploads to avoid random? failures from google cloud
-          // @see https://stackoverflow.com/questions/53172050/google-cloud-storage-invalid-upload-request-error-bad-request
-          resumable: false
+          resumable: this.config.resumable
         });
         resolve(null);
       } catch (err) {

--- a/src/types/index.ts
+++ b/src/types/index.ts
@@ -12,6 +12,8 @@ export interface VerdaccioConfigGoogleStorage extends Config {
   keyFilename?: string;
   // disable bucket validation
   validation?: GoogleValidation;
+  /** Enable/disable resumable uploads to GC Storage */
+  resumable?: boolean;
 }
 
 export type GoogleValidation = boolean | string;

